packages/keymaps/src/browser/keybindings-widget.tsx

```diff
@@ -0,0 +1,268 @@
+/********************************************************************************
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import React = require('react');
+import * as fuzzy from 'fuzzy';
+import { injectable, inject, postConstruct } from 'inversify';
+import { CommandRegistry, Command, MessageService } from '@theia/core/lib/common';
+import { ReactWidget } from '@theia/core/lib/browser/widgets/react-widget';
+import { KeybindingRegistry, SingleTextInputDialog } from '@theia/core/lib/browser';
+
+export interface KeybindingItem {
+    id: string,
+    command: string,
+    keybinding?: string,
+    context?: string,
+}
+
+export interface CellData {
+    value: string,
+    highlighted: boolean,
+}
+
+@injectable()
+export class KeybindingWidget extends ReactWidget {
+
+    @inject(CommandRegistry)
+    protected readonly commandRegistry: CommandRegistry;
+
+    @inject(KeybindingRegistry)
+    protected readonly keybindingRegistry: KeybindingRegistry;
+
+    @inject(MessageService)
+    protected readonly messageService: MessageService;
+
+    protected items: KeybindingItem[];
+
+    static readonly ID = 'keybindings.view.widget';
+
+    protected query: string = '';
+
+    protected readonly fuzzyOptions = {
+        pre: '<match>',
+        post: '</match>',
+    };
+
+    constructor() {
+        super();
+        this.id = KeybindingWidget.ID;
+        this.title.label = 'Keyboard Shortcuts';
+        this.title.closable = true;
+        this.update();
+    }
+
+    @postConstruct()
+    protected init(): void {
+        this.items = this.getItems();
+    }
+
+    protected searchKeybindings = () => {
+        this.items = [];
+        const searchField = this.findSearchField();
+        this.query = searchField ? searchField.value.trim().toLocaleLowerCase() : '';
+        const queryEncoded = encodeURI(this.query).toString();
+        const items = this.getItems();
+        items.forEach(item => {
+            const keys: (keyof KeybindingItem)[] = ['id', 'command', 'context', 'keybinding'];
+            let matched = false;
```

Comment:
my only concern is that if I early exit for instance because a value in `command` matched, other columns for that same row won't get fuzzy highlighting.

Reply:
ok, makes sense